pkg/storegateway/gateway_http.go

```diff
@@ -0,0 +1,53 @@
+package storegateway
+
+import (
+	"net/http"
+	"text/template"
+
+	"github.com/go-kit/kit/log/level"
+
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/cortexproject/cortex/pkg/util/services"
+)
+
+var (
+	statusPageTemplate = template.Must(template.New("main").Parse(`
+	<!DOCTYPE html>
+	<html>
+		<head>
+			<meta charset="UTF-8">
+			<title>Cortex Store Gateway Ring</title>
+		</head>
+		<body>
+			<h1>Cortex Store Gateway Ring</h1>
+			<p>{{ .Message }}</p>
+		</body>
+	</html>`))
+)
+
+func writeMessage(w http.ResponseWriter, message string) {
+	w.WriteHeader(http.StatusOK)
+	err := statusPageTemplate.Execute(w, struct {
+		Message string
+	}{Message: message})
+
+	if err != nil {
+		level.Error(util.Logger).Log("msg", "unable to serve store gateway ring page", "err", err)
+	}
+}
+
+func (c *StoreGateway) RingHandler(w http.ResponseWriter, req *http.Request) {
```

Comment:
We have multiple copies of this code around Cortex now, we should unify it into single "ring handler" (again, in separate PR)

Reply:
Agree. Will take care in a separate PR.